### DISCUSSION TITLE
Why does an update of a package on an "unsupported  system" trigger rebuilds?

### DISCUSSION THREAD

**Discussion by therealprof** [Homebrew/discussions#2556] [Everyday usage]
Why do upgrades (even of casks) force an up-to-date check and update of all dependencies? `--skip-cask-deps` also doesn't seem to have any effect. That seems very gratuitous and conflict prone to me and in fact is causing me troubles exactly due to the "You will encounter build failures with some formulae." part of the big fat (and rather annoying!) warning.

E.g. `plantuml` has a dependency to `openjdk`(any version) which is installed but not "up-to-date":
```
openjdk (17) < 17.0.1
```
Since the upgrade doesn't work and brew refuses to do anything for `plantuml` I'm stuck. Pinning it also doesn't provide any useful outcome:
```
==> Upgrading 1 outdated package:
plantuml 1.2021.12 -> 1.2021.14
Error: You must `brew unpin openjdk` as installing plantuml requires the latest version of pinned dependencies
```

I've already spent half a CPU day recompiling a ton of software which was working just fine and it seems rather wasteful (also to the environment!) to require frequent recompilation of whole compilers and other expensive libraries (like qt) just to being able to keep software up to date.

I understand you don't want to support "old" versions of macOS (although there're perfectly valid reasons not to upgrade!) and thus not provide any prepacked software, but I don't quite see why unsupported systems are forced to reocmpile and upgrade all working software with the latest and greatest unsupported software just to compile other unsupported software.

**Comment by carlocab** [comment]
We're tweaking this behaviour in https://github.com/Homebrew/brew/pull/12487.

It's not going to help with `plantuml` specifically (since `openjdk` is a direct dependency), but this should hopefully limit the number of software you have to rebuild when there are no bottles available.
